Jenkinsfile

```diff
@@ -194,6 +222,12 @@ node() {
 
             sh "curl -u ${env.USER_PASS} -d '${json}' -H 'Content-Type: application/json' -X POST ${env.PR_STATUS_URI}"
 
+            sh '''
```

Comment:
We may want to leave the projects around in case of error so the issue can be looked ... or at least have the option to not delete.

Reply:
I thought part of the point of this issue was to prevent there being loads of projects left around?

That being said, I can see your point and perhaps leaving it intact on failed attempts.

Reply:
Yes, we need to ensure the system(s) don't fill up with old projects. However, we also need to ensure that issues can be debugged. A few alternatives (and there's probably more):
1) Set up the [cronjob](https://github.com/redhat-cop/openshift-management/blob/master/jobs/cronjob-prune-projects.yaml) I mentioned before to do periodic clean-up of projects - just like we do for builds, images, etc. We just need to define what projects are on the "safe list" to avoid the wrong projects get removed. (may need an enhancement to the prune job to only remove projects matching for example a regex)
2) provide an option for the CI job to leave projects around on failure so the user can re-run with the option set and hence don't have the projects removed (user will have to manually do the remove after they are done)
3) schedule a self-destruct job for failed projects with it being cleaned up after X hours (this is new functionality)